types/eonasdan-bootstrap-datetimepicker/index.d.ts

```diff
@@ -1209,6 +1209,7 @@ declare global {
     interface JQuery {
         datetimepicker(): JQuery;
         datetimepicker(options: SetOptions): JQuery;
+        datetimepicker(methodName: string, ...args: any[]): JQuery | any;
```

Comment:
Add
```ts
datetimepicker(methodName: "destroy"): JQuery;
datetimepicker(methodName: "hide"): JQuery;
datetimepicker(methodName: "show"): JQuery;
datetimepicker(methodName: "toggle"): JQuery;
```
Above this, then this line is fine - this should let chaining work with type inference, but still allow calling arbitrary methods.